misc/identity-keys/src/identity-keys.ts

```diff
@@ -52,6 +53,40 @@ export class IdentityKeys implements IIdentityKeys {
     return this.identityKeys.keys.includes(account);
   }
 
+  public async prepareRegistrationViaRecaps({
+    domain,
+    recapObject
+  }: {
+    domain: string
+    recapObject: {
+      att: Record<string,any>
+    }
+  }) {
+    const encodedRecap = objectToHex(recapObject);
+    const authRecap = `urn:recap:${encodedRecap}`
+
+    const { privKeyHex, pubKeyHex, privateKey } = await this.generateIdentityKey();
+    const didKey = encodeEd25519Key(privKeyHex);
+
+    const uri = `${domain}?walletconnect_identity_token=${didKey}`
```

Comment:
This needs to include the scheme. I think we should change `domain` to be `url: URL` and then parse the domain out of it and add the query param here with a URL builder.

Also change `walletconnect_identity_token` to `walletconnect_identity_key`